engine/graphics/src/graphics.cpp

```diff
@@ -300,8 +324,21 @@ namespace dmGraphics
     }
     bool Initialize()
     {
-        return SelectGraphicsAdapter() && g_functions.m_Initialize();
+        // Select by priority
+        return SelectGraphicsAdapterByPriorityOrType(true, ADAPTER_TYPE_NULL) && g_functions.m_Initialize();
```

Comment:
What about something like this?

```
bool Initialize(const char* adapter_name) {
    if (g_Adapter)
        return true;
    
    Adapter adapter;
    bool result = GetAdapterFromString(adapter_name, &adapter);
    if (result) {
        result = SelectAdapterByType(adapter);
    }
    if (!result)
        result = SelectAdapterByPriority();
    
    if (result)
        return g_functions.m_Initialize();
    return false;
}
```

I think it removes some of the configuration stuff from engine.cpp, which I think might be good.